capsules/src/net/udp/driver.rs

```diff
@@ -364,34 +289,64 @@ impl<'a> LegacyDriver for UDPDriver<'a> {
         &self,
         appid: AppId,
         allow_num: usize,
-        slice: Option<AppSlice<SharedReadWrite, u8>>,
-    ) -> ReturnCode {
-        match allow_num {
-            0 | 1 | 2 | 3 => self.do_with_app(appid, |app| {
-                let mut success = true;
-                match allow_num {
-                    0 => app.app_read = slice,
-                    1 => match slice {
```

Comment:
I'd suggest compacting the identifier space.

Reply:
done.